### PR TITLE
Make `headless_defaults` example actually headless

### PR DESCRIPTION
**Pull request by LegNeato**
Fixes https://github.com/bevyengine/bevy/issues/5260

### CODE REVIEW

File: examples/app/headless_defaults.rs [hunk 1]
```diff
@@ -1,14 +1,29 @@
...
         .add_plugins(DefaultPlugins)
+        .add_system(do_something)
         .run();
 }
+
+// Normally Bevy exits when all windows are closed.
+//
+// When running in headless mode there are no windows so
+// you must manually send an [`bevy::app::AppExit`] event.
+fn do_something(mut app_exit_events: EventWriter<AppExit>) {
```
Comment (DJMcNab) [inline examples/app/headless_defaults.rs:26]: `do_something` is not an idiomatic system name. I think something like `exit_unconditionally` would be better.

**Review by DJMcNab — CHANGES_REQUESTED** [review]
I don't really like this change - it misses that the point of the example is to test running a complete bevy application without the renderer, and to ensure we support that. This capability would be useful if alternative renderers are being used.

I'm not quite sure on the reason for the headless mode existing aside from just disabling the `render` feature (although testing it would require more set-up)

**Comment by LegNeato** [comment]
Well, the example is not what one would commonly consider headless...so perhaps the change here is to rename it if the behavior is intended?

**Comment by LegNeato** [comment]
Abandoning this in favor of renaming

**Pull request closed without merging**